.github/workflows/multibuild.yaml

```diff
@@ -158,8 +158,34 @@ jobs:
           path: ./packages/dart/tarballs/${{ matrix.output-name }}.tgz
           if-no-files-found: error
 
+  universal_sh:
+    if: startsWith(github.ref, 'refs/tags/v')
+    defaults:
+      run:
+        working-directory: ./packages/dart/sshnoports/bundles
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - run: |
+          write_metadata() {
+            start_line="# SCRIPT METADATA"
+            end_line="# END METADATA"
+            file=$1
+            variable=$2
+            value=$3
+            # since this is linux only, sed -i is safe without a file ext.
+            sed -i "/$start_line/,/$end_line/s|$variable=\".*\"|$variable=\"$value\"|g" "$file"
+          }
+          REF=${{ github.ref }}
+          TAG=${REF:11}
+          write_metadata universal.sh sshnp_version "$TAG"
```

Comment:
If the action is called from a tag, then it will replace the version in the universal.sh script with the version from the tag, then it will upload that as an artifact.